docs/src/content/en/docs/agents/dynamic-agents.mdx

```diff
@@ -0,0 +1,47 @@
+---
+
+title: Dynamic Agents | Agents | Mastra Docs
+description: Dynamically configure your agent's instruction, model and tools using runtime context.
+
+# Dynamic Agents
+
+Dynamic agents use [runtime context](./runtime-variables), like user IDs and other important parameters, to adjust their settings in real-time. This means they can change the model they use, update their instructions, and select different tools as needed. By using this context, agents can better respond to each user's needs. They can also call any API to gather more information, which helps improve what the agents can do.
+
+## Key Benefits
```

Comment:
would remove the key benefits section (since we can cover this in opening couple sentences)